code/modules/reagents/reagent_containers/food/condiment.dm

```diff
@@ -14,6 +14,16 @@
 	possible_transfer_amounts = list(1,5,10)
 	center_of_mass = list("x"=16, "y"=6)
 	volume = 50
+	var/next_shake
+
+/obj/item/reagent_containers/food/condiment/proc/shake(var/mob/user)
+	if(world.time >= next_shake)
+		if(reagents.total_volume > 0)
+			user.visible_message(pick(SPAN_NOTICE("[user] shakes \the [src]."), SPAN_NOTICE("[user] gives \the [src] a good shake.")), SPAN_NOTICE("You give \the [src] a good shake."))
+			playsound(get_turf(src),'sound/items/condiment_shaking.ogg', rand(10,50), 1)
+		else
+			user.visible_message(pick(SPAN_NOTICE("[user] shakes \the [src], but it makes no noise."), SPAN_NOTICE("[user] gives \the [src] a good shake, but it makes no noise.")), SPAN_NOTICE("You give \the [src] a good shake, but it makes no noise."))
+	next_shake = world.time + 20
```

Comment:
I did just notice this so I apologize for not putting it in an earlier review. This is an optional change.
But what this will do is always set the next_shake to be 2 seconds in the future, even if we did not pass the `if(world.time >= next_shake)` check. So, I shake. One second later, I try to shake again. As intended, it does not display the message. However, now the next shake is 2 seconds from THAT attempt. So, if I keep attempting every second, it will never go.

This is fine, but if you want it to be 2 seconds from the last successful shake without resetting the timer on failed ones, just move the `next_shake = world.time + 20` in by one more indent, so it is caught under the world.time >= next_shake check.